File: Exemplos/Java/src/main/java/com/stefanini/lampada/Network.java
package com.stefanini.lampada;

import kong.unirest.Unirest;

public class Network {

	private final String PATH_BASE = "http://SOALV3DFGC01:8080/hack/";
	private final int ID_EQUIPE = 0;

	public Network() {}

	public LampadaPutResponseBody setLampadaStatus(Integer codLampada, boolean status) {
		String url = String.format("%s%s", PATH_BASE, "api/lampada/{codLampada}/status");
		return Unirest.put(url)
				.header("Content-Type", "application/json")
				.routeParam("codLampada", codLampada.toString())
				.body(new LampadaPutRequestBody(ID_EQUIPE, status))
				.asObject(LampadaPutResponseBody.class).getBody();
	}

	public class LampadaPutRequestBody {
		public int idEquipe;
		public boolean status;

		public LampadaPutRequestBody(int idEquipe, boolean status) {
			this.idEquipe = idEquipe;
			this.status = status;
		}
	}

	public class LampadaPutResponseBody {
		public int lampada;
		public String status;

		@Override
		public String toString() {
			return "Lampada " + lampada + " -> " + status;
		}
	}
}

File: Exemplos/Python/service/network.py
import requests, os
import json
class Network:
    @staticmethod
    def _api(lampadaId, status):
        url = "http://soalv3dfgc01:8080/hack/api/lampada/%i/status" % lampadaId
        payload = {
            "status": status,
            "idEquipe": os.environ["ID_EQUIPE"]
        }
        headers = {
        'Content-Type': 'application/json'
        }
        response = requests.request("PUT", url, headers = headers, data = json.dumps(payload))
        print(response.text.encode('utf8'))

    @staticmethod
    def atualizaStatus(lampadaId, status):
        Network._api(lampadaId, status)